### ISSUE TITLE
Feature Request: add LLava as image recognizer on web pages

### ISSUE THREAD

**Issue by amonpaike** [n4ze3m/page-assist#33]
It would be nice to add LLava to recognize images in web pages so that it works together with RAG models..  LLava is however a textual AI model so it can be used as the main model to describe pages, this is to avoid overloading loading times. you just need to be able to "see" the various images.

On hugging face there are also models like mistral7B mixed with llava....

**Comment by amonpaike** [comment]
LLAVA   already works if you copy and paste the image link into the web page ,,,,

![image](https://github.com/n4ze3m/page-assist/assets/884282/c3e1f225-d68f-44c4-8282-c02934335753)

**Comment by n4ze3m** [comment]
Yes, both the web UI and the side panel support uploading images.

![image](https://github.com/n4ze3m/page-assist/assets/39720973/924eb462-ab83-4931-a091-1f691c406f5a)

There are a few more side panel supports for:

- Chat with YouTube video (with transcript)
- Chat with a PDF opened on a website

**Comment by amonpaike** [comment]
Nice!  I assume that it will therefore be much less complicated to make the RAG AI model interact with the LLAVA description of the images to create a single summary analysis.  I hope this is the case, it will be a wonderful feature.

**Comment by amonpaike** [comment]
I think that the work you will have to do is even less complicated, already now, if I open an image from the web as a page LLAVA can see it and describe it.. At this point the only thing you will have to do is indicate the links to LLAVA of the relevant images in the page and mix them with the text that the RAG provides to the AI.. :) it seems easy, but obviously I don't know anything about it, it's just a guess.

![image](https://github.com/n4ze3m/page-assist/assets/884282/0b24ec34-1c42-49a0-8058-c7cabf96bd71)

**Comment by amonpaike** [comment]
More fun! In reality the AI with the RAG it is already capable of extracting the link of the images! They just need to be sent to LLAVA to write the description of the image.
![image](https://github.com/n4ze3m/page-assist/assets/884282/6b95e421-48f6-42cf-8c6b-99c6b92b2758)

**Comment by n4ze3m** [comment]
Yes, it's possible, I guess. I will add a setting to configure the model with vision for the image URL in the upcoming release :)

**Comment by amonpaike** [comment]
bingo mate!, now is your turn to do the magic here... 
LLAVA  whit the RAG is already capable to extract the link of the images and describing it .... 
(all you have to do is invoke LLAVA like you did for the RAG and mix everything..) 

![image](https://github.com/n4ze3m/page-assist/assets/884282/614342eb-4ad3-4045-8c44-0602383166b6)

**Comment by n4ze3m** [comment]
Thank you for the suggestions. We will improve on them in the upcoming version. By the way, Page Assist has a ChatGPT-like web UI. Have you tried it? If not, click on the Page Assist icon.

The web UI supports internet search and many more features. :)

![image](https://github.com/n4ze3m/page-assist/assets/39720973/d6079bc3-848c-4e8a-8452-22708b80b7ad)
![image](https://github.com/n4ze3m/page-assist/assets/39720973/5e5b5f8c-3189-42f5-b7cf-10e41fb6255e)

**Comment by amonpaike** [comment]
yes I tried it a little, but at the moment I'm excited by the concept of the private co-pilot next to the web pages
